ts/src/header-validator/validate-json.ts

```diff
@@ -1361,6 +1421,44 @@ function eventTriggerData(
   )
 }
 
+function positiveUint32(ctx: Context, j: Json): Maybe<number> {
+  return number(ctx, j)
+    .filter((n) => isInteger(ctx, n))
+    .filter((n) => isInRange(ctx, n, 1, UINT32_MAX))
+}
+
+function maxEventStates(ctx: Context, j: Json): Maybe<number> {
```

Comment:
This function doesn't match the corresponding specification algorithm, as this one requires the field to be a JSON number, whereas the spec requires the field to be a string formatted as a base-10 integer.

If the spec is correct, then this function should just be replaced with `int64` or `uint64`.

Reply:
Fixed the spec to be consistent.

Reply:
This didn't fix the inconsistency. The fundamental discrepancy between JSON string and JSON number is still present.

Reply:
Done. Note I kept this function so that we can customize the range.